File: ios/GitaVani/GitaVani/Models/LocalizedText.swift
import Foundation

struct LocalizedText: Codable {
    let en: String
    let hi: String

    func forLanguage(_ language: String) -> String {
        language == "hindi" ? hi : en
    }

    func oppositeLanguage(_ language: String) -> String {
        language == "hindi" ? en : hi
    }
}

File: android/GitaVani/app/src/main/java/com/nikhilsi/gitavani/model/LocalizedText.kt
package com.nikhilsi.gitavani.model

import kotlinx.serialization.Serializable

@Serializable
data class LocalizedText(
    val en: String,
    val hi: String
) {
    fun forLanguage(language: String): String =
        if (language == "hindi") hi else en

    fun oppositeLanguage(language: String): String =
        if (language == "hindi") en else hi
}
